packages/realm_dart/lib/src/list.dart

```diff
@@ -42,8 +42,12 @@ abstract class RealmList<T extends Object?> with RealmEntity implements List<T>
   /// Creates a frozen snapshot of this `RealmList`.
   RealmList<T> freeze();
 
+  //TODO Should we have a base collection class so we can move those methods there?
   /// Allows listening for changes when the contents of this collection changes.
   Stream<RealmListChanges<T>> get changes;
+
+  /// Allows listening for changes when the contents of this collection changes on one of the provided [keyPaths].
+  Stream<RealmListChanges<T>> changesFor([List<String>? keyPaths]);
```

Comment:
That would actually make sense. 
The only "issue" I see is that now `changes` is defined as `changesFor(null)` to avoid repetitions. 
If we move `changesFor` to an extension method then we'd need to have some duplication (admittedly not a biggie though). It also seems Dart does not have protected methods, so I don't know if I can go around this with some other language constructs. Any suggestion?

Reply:
You could make a private `_changesFor` that both the `changesFor` extension method, and `changes` member delegates to?

Reply:
Yeah, I was thinking something like that. In my mind it sounded uglier than it would probably be, I'll give it a try 😁

Reply:
I've done it in the last commit